data/spec/SBOM.md

```diff
@@ -1,3 +1,201 @@
+      ]
+    },
+    {
+      "type": "sbom",
+      "accept_scan_request": "application/vnd.scanner.adapter.scan.request+json; version=1.1",
+      "consumes_mime_types": [
+        "application/vnd.oci.image.manifest.v1+json",
+        "application/vnd.docker.distribution.manifest.v2+json"
+      ],
+      "produces_mime_types": [
+        "application/vnd.security.sbom.report+json; version=1.0"
+      ]
+    }
+  ],
+```
+
+### Scan SBOM
+
+The capability to scan the image vulnerabilities from the SBOM of image, which have the better performance as the scanner only needs to pull the SBOM artifact instead of whole image. The basic process is exactly same with the image vulnerabilities scan because the SBOM has been bundled as an [OCI artifact](#harbor-sbom-artifact-layout). If the `artifactType` is `application/vnd.goharbor.harbor.sbom.v1` of the manifest, then scanner should treat is as SBOM artifact and pull the raw SBOM content from layers and scan vulnerabilities from it, the `artifact` field in the report should be the **subject** artifact.
+
+### Harbor SBOM Artifact Layout
+
+The support of [OCI Distribution spec v1.1's](https://github.com/goharbor/community/blob/main/proposals/new/distribution-1.1-adoption.md) referrer API in harbor enables the packaging of SBOM into OCI artifacts and their association with the respective subject artifact based on its attributes. Subsequently, by following the OCI distribution v2 image push process and API, you can push the SBOM artifact to Harbor, where it will be automatically processed as an accessory to the subject artifact. The layout is designed by following the [artifact guidelines](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc5/manifest.md#guidelines-for-artifact-usage).
+
+Here are some constraints:
+
+1. The `artifactType` MUST set to **application/vnd.goharbor.harbor.sbom.v1** because harbor leverages this to identify it as the SBOM, and set `config` to the [empty descriptor value](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc5/manifest.md#guidance-for-an-empty-descriptor).
+2. The SBOM file should be packed as the artifact layer, should not be compressed, the `mediaType` should set the format of the SBOM file, currently only **application/spdx+json** and **application/vnd.cyclonedx+json** are supported. (These mediaType are registered on the [IANA](https://www.iana.org/assignments/media-types/media-types.xhtml))
+3. The vendor can add the customize attributes to the `annotations`.
+4. The `subject` should set the info of the subject artifact.
+5. The `layers` should only contains one layer which storing the SBOM file, you should separate to multiple SBOM artifact if you have multiple SBOM files with different formats.
+
+Example
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+  "artifactType": "application/vnd.goharbor.harbor.sbom.v1",
```

Comment:
I took another look at [the OCI artifact spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage). Is it allowed that the manifest has a different value in `artifactType` from `layers[].mediaType`?

>Does the artifact have additional JSON formatted metadata as configuration? If yes, continue to 3. If no, specify the artifactType, include the artifact in the layers, and set config to the empty descriptor value. Here is an example of this with a single layer:

I presumed they should match. The example has the same value (`application/vnd.oci.image.manifest.v1+json`).

Reply:
There is no mandatory requirement to specify the `artifactType` and layer `mediaType` must be the same, they are two concepts, one is used to represent the entire type of the artifact, the other is used to represent the data in each layer of the specific type, to the container image, for example, the artifact type is image, and each layer of the type may be different, such as tar or gzip.

Reply:
It would be okay if we ensure the Harbor spec isn't violating the OCI spec.